"""
ç®€å•å¯¹è¯æ¡†ä¸»ç¨‹åº - æ¼”ç¤ºå¦‚ä½•ä½¿ç”¨ Designer è®¾è®¡çš„ç•Œé¢

å·¥ä½œæµç¨‹:
1. simple_dialog.ui - åœ¨ Designer ä¸­è®¾è®¡
2. simple_dialog.py - ä½¿ç”¨ pyuic6 è½¬æ¢ç”Ÿæˆ(ä¸è¦æ‰‹åŠ¨ä¿®æ”¹!)
3. simple_dialog_main.py - æœ¬æ–‡ä»¶,æ·»åŠ ä¸šåŠ¡é€»è¾‘å’Œäº‹ä»¶å¤„ç†
"""
from PyQt6.QtWidgets import QApplication, QDialog
import sys

# å¯¼å…¥ pyuic6 è½¬æ¢ç”Ÿæˆçš„ UI ç±»
from simple_dialog import Ui_Dialog


class SimpleDialog(QDialog):
    """ç®€å•å¯¹è¯æ¡† - æ¼”ç¤ºä¿¡æ¯è¾“å…¥å’Œæ˜¾ç¤º"""
    
    def __init__(self):
        super().__init__()
        
        # åˆ›å»º UI å¯¹è±¡å¹¶è®¾ç½®åˆ°å½“å‰çª—å£
        self.ui = Ui_Dialog()
        self.ui.setupUi(self)
        
        # è¿æ¥ä¿¡å·å’Œæ§½
        self.connect_signals()
    
    def connect_signals(self):
        """è¿æ¥æ§ä»¶çš„ä¿¡å·åˆ°å¤„ç†å‡½æ•°"""
        # æ˜¾ç¤ºä¿¡æ¯æŒ‰é’®
        self.ui.btn_show.clicked.connect(self.show_info)
        
        # æ¸…ç©ºæŒ‰é’®
        self.ui.btn_clear.clicked.connect(self.clear_info)
        
        # å…³é—­æŒ‰é’®
        self.ui.btn_close.clicked.connect(self.close)
        
        # å§“åè¾“å…¥æ¡†å›è½¦é”®
        self.ui.txt_name.returnPressed.connect(self.show_info)
    
    def show_info(self):
        """æ˜¾ç¤ºç”¨æˆ·è¾“å…¥çš„ä¿¡æ¯"""
        # è·å–è¾“å…¥çš„å€¼
        name = self.ui.txt_name.text()
        age = self.ui.spin_age.value()
        gender = self.ui.combo_gender.currentText()
        
        # éªŒè¯è¾“å…¥
        if not name.strip():
            self.ui.txt_info.setPlainText("âš ï¸ è¯·è¾“å…¥å§“å!")
            return
        
        # æ ¼å¼åŒ–æ˜¾ç¤º
        info = f"""
ğŸ“‹ ç”¨æˆ·ä¿¡æ¯
{'='*30}
å§“å: {name}
å¹´é¾„: {age} å²
æ€§åˆ«: {gender}
{'='*30}
        """
        
        self.ui.txt_info.setPlainText(info.strip())
    
    def clear_info(self):
        """æ¸…ç©ºæ‰€æœ‰è¾“å…¥"""
        self.ui.txt_name.clear()
        self.ui.spin_age.setValue(20)
        self.ui.combo_gender.setCurrentIndex(0)
        self.ui.txt_info.clear()
        
        # ç„¦ç‚¹å›åˆ°å§“åè¾“å…¥æ¡†
        self.ui.txt_name.setFocus()


def main():
    """ä¸»å‡½æ•°"""
    app = QApplication(sys.argv)
    
    # åˆ›å»ºå¹¶æ˜¾ç¤ºå¯¹è¯æ¡†
    dialog = SimpleDialog()
    dialog.show()
    
    sys.exit(app.exec())


if __name__ == '__main__':
    main()
